### ISSUE TITLE
[feature request] [ux] Introduce `SDPBackend.ALL` (or similar functionality) to mean "all available backends"

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#137486]
### 🚀 The feature, motivation and pitch

This would simplify code using `torch.nn.attention.sdpa_kernel` as the list of backends may evolve (e.g. Flex Attention might become a backend for SDPA?) and hardcoding the list of all available backends in end-user code is brittle.

Otherwise some hacks required as supporting two code-paths: one with `sdpa_kernel` context manager, another without, and this complicates the code / graph.

Alternative solutions might be:
- supporting `with torch.nn.attention.sdpa_kernel([]):` to mean all backends
- supporting SDPBackend.all_backends or something to provide a list of all backends

Another minor incovenience is that `torch.nn.attention.sdpa_kernel(...)` accepts only a `list[SDPBackend]` while in configs often a list of strings is simpler to specify, so preprocessing with `getattr` is currently needed

**Comment by drisspg** [comment]
By default (without using the context manager)  all available backends are enabled

**Comment by vadimkantorov** [comment]
Yes, my point is that having two code paths for specified backends and another code path (or coming up with hacks like trying to find all backends from `SDPBackend.__members__` and subtracting SDPBackend.ERROR) for all backends complicates the user code, so having a special value for meaning all backends would allow to always use context manager without hacks
